firestore-multimodal-genai/functions/src/generative-client/generative_ai.ts

```diff
@@ -39,7 +39,13 @@ export class GeminiGenerativeClient extends GenerativeClient<
         throw new Error('Gemini Pro Vision selected, but missing Image Field');
       }
 
-      const base64String = await getImageBase64(options.image, 'google-ai');
+      let base64String: string;
+
+      if (options.image.startsWith('gs://')) {
+        base64String = await getImageBase64(options.image, 'google-ai');
+      } else {
+        base64String = options.image.replace(/^data:image\/\w+;base64,/, '');
```

Comment:
The only issue i see here is that getImageBase64 was compressing the image based on the API provider, this string could be larger than the API limit.

To compress the string we'd probably have to use sharp or something to render it in a format i guess?

Reply:
Added compression logic for base64 string images.